BinaryOracle/contracts/Oracle.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+import "./interfaces/IOracle.sol";
+
+contract Oracle is AccessControl, IOracle {
```

Comment:
Please order functions layout by type
Refer https://docs.soliditylang.org/en/v0.8.17/style-guide.html#order-of-layout

Recommendation Order:
State Variables
Events
Modifiers
Constructor
External functions
External functions that are view
External functions that are pure
Public functions
Internal functions
Private functions